message_ix/model/MESSAGE/model_core.gms

```diff
@@ -583,6 +588,47 @@ COMMODITY_BALANCE(node,commodity,level,year,time)$( map_commodity(node,commodity
 %SLACK_COMMODITY_BALANCE%   - SLACK_COMMODITY_BALANCE_LO(node,commodity,level,year,time)
     =G= 0 ;
 
+* Equation COMMODITY_BALANCE_FULL
```

Comment:
It may at this point be easier to have an equation `COMMODITY_EQUIVALENCE` which equates the commodity balance to an auxiliary variable (similar to the variable `REL`), and then have much shorter lower and upper bounds on that.

Reply:
I think this is a good idea. So we don't have to worry about exclusion-purpose sets. Also any possible future case of using only LT, EQ or GT balances are embedded.